app/controllers/api/contacts_controller.rb

```diff
@@ -0,0 +1,14 @@
+class Api::ContactsController < ApplicationController
+	def index
```

Comment:
I suspect the performance issues here. Can you help me in identifying them and suggest a fix ?